specification/extendedlocation/resource-manager/Microsoft.ExtendedLocation/preview/2021-08-31-preview/customlocations.json

```diff
@@ -0,0 +1,1196 @@
+            "type": "object",
+            "properties": {
+              "apiVersion": {
+                "description": "Api Version of Resource Type",
+                "type": "string"
+              },
+              "resourceProviderNamespace": {
+                "description": "Resource Provider Namespace of Resource Type",
+                "type": "string"
+              },
+              "resourceType": {
+                "description": "Resource Type",
+                "type": "string"
+              }
+            }
+          }
+        }
+      }
+    },
+    "EnabledResourceTypesListResult": {
+      "description": "List of EnabledResourceTypes definition.",
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "The URL to use for getting the next set of results.",
+          "type": "string",
+          "readOnly": true
+        },
+        "value": {
+          "description": "The list of EnabledResourceTypes available for a customLocation.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/EnabledResourceType"
+          }
+        }
+      }
+    },
+    "Identity": {
+      "description": "Identity for the resource.",
+      "type": "object",
+      "properties": {
+        "principalId": {
+          "description": "The principal ID of resource identity.",
+          "type": "string",
+          "readOnly": true
+        },
+        "tenantId": {
+          "description": "The tenant ID of resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "The identity type.",
+          "type": "string",
+          "enum": [
+            "SystemAssigned",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "ResourceIdentityType",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "customLocation": {
+      "description": "Custom Locations definition.",
+      "type": "object",
+      "properties": {
+        "identity": {
+          "$ref": "#/definitions/Identity",
+          "description": "Identity for the resource."
+        },
+        "properties": {
+          "$ref": "#/definitions/customLocationProperties",
+          "description": "The set of properties specific to a Custom Location",
+          "x-ms-client-flatten": true
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "description": "Metadata pertaining to creation and last modification of the resource",
+          "readOnly": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/TrackedResource"
+        }
+      ]
+    },
+    "customLocationFindTargetResourceGroupProperties": {
```

Comment:
why can't this be strongly typed instead of a dynamic dictionary?

Reply:
Updated the description. This is the labels of the custom resource, it's a dynamic dictionary in Kubernetes so we reuse the same concept here.